src/Psl/Regex/replace_with.php

```diff
@@ -15,7 +15,7 @@
  * `$callback`.
  *
  * @param non-empty-string $pattern The pattern to search for.
- * @param (callable(array<int, string>): string) $callback The replacement callable.
+ * @param (callable(string[]): string) $callback The replacement callable.
```

Comment:
```suggestion
 * @param (callable(array<array-key, string>): string) $callback The replacement callable.
```

Reply:
we don't use phpstorm generics here :)

Reply:
`string[]` and `array<array-key, string>` are exactly the same: https://psalm.dev/r/212f934d6c

Do you still prefer the more verbose version?

Reply:
`string[]` is actually a valid array type according to the proposed PSR-5 PHPDoc: https://github.com/php-fig/fig-standards/blob/d82fd79a13ab3cd2676ed65dafbf4196de9fd322/proposed/phpdoc.md#abnf

Reply:
yes, we don't use `T[]` syntax in PSL